### PR TITLE
move waived param to yaml

### PR DESCRIPTION
**Pull request by wizardofoz123**
This requires complementary kpet-db change.

Signed-off-by: Jakub Racek <jracek@redhat.com>

### CODE REVIEW

File: kpet/data.py [hunk 1]
```diff
@@ -283,6 +288,7 @@ def convert(old_data):
                         kickstart=String(),
                         match=Class(PositivePattern),
                         dont_match=Class(NegativePattern),
+                        fetch_suffix=String(),
```
Comment (spbnick) [inline kpet/data.py]: Could you please rename this to `url_suffix`? That would be clearer, I think.

File: kpet/data.py [hunk 2]
```diff
@@ -266,13 +269,15 @@ def convert(old_data):
...
                     )
                 ),
                 convert,
                 Struct(
                     required=dict(
                         description=String(),
-                        cases=List(Class(Case))
+                        cases=List(Class(Case)),
+                        maintainer=String(),
```
Comment (spbnick) [inline kpet/data.py]: Some tests have multiple maintainers, so could you please name it `maintainers` and make it a list of strings instead? Also, please make it optional with an empty list as the default (replacing `None` with `[]` right here in the constructor). This way we also won't have to update so many test assets.

File: kpet/data.py [hunk 3]
```diff
@@ -266,13 +269,15 @@ def convert(old_data):
                         hostRequires=String(),
                         partitions=String(),
                         kickstart=String(),
+                        fetch_suffix=String(),
```
Comment (spbnick) [inline kpet/data.py]: Please don't update the legacy schema. First, we shouldn't change the past, and then we need to remove support for it (and update the tests/assets) at the first opportunity, since the database has migrated away.

File: tests/assets/db/general/trees/rhel7.xml [hunk 4]
```diff
@@ -1,15 +1,39 @@
...
   {% for suite in host.suites %}
     {% if suite[template_field] %}
       {% include suite[template_field] %}
     {% endif %}
     {% for case in suite.cases %}
-      {% if case[template_field] %}
-        {% include case[template_field] %}
-      {% endif %}
+        {% if template_field == 'tasks' %}
+            {% if case['role'] is not none %}
```
Comment (spbnick) [inline tests/assets/db/general/trees/rhel7.xml]: Casserole :shallow_pan_of_food:, yummy :grin:!

File: tests/assets/db/general/trees/rhel7.xml [hunk 5]
```diff
@@ -1,15 +1,39 @@
...
+                    {% if case['extra_params'] is not none %}
+                      {{ case['extra_params'] }}
+                    {% endif %}
+                  </params>
+            </task>
+        {% else %}
+            {% if case[template_field] %}
+                {% include case[template_field] %}
+            {% endif %}
+        {% endif %}
```
Comment (spbnick) [inline tests/assets/db/general/trees/rhel7.xml]: I'll leave some comments about this part in the kpet-db change. Please update here as well.

File: tests/assets/db/general/trees/rhel7.xml [hunk 6]
```diff
@@ -55,10 +79,10 @@
           </params>
         </task>
         <task name="/test/misc/machineinfo">
-          <fetch url="https://github.com/CKI-project/tests-beaker/archive/staging.zip#test/misc/machineinfo"/>
+          <fetch url="https://github.com/CKI-project/tests-beaker/archive/master.zip#test/misc/machineinfo"/>
```
Comment (spbnick) [inline tests/assets/db/general/trees/rhel7.xml:84]: Please make the URL changes a separate commit.

File: tests/assets/db/general/suites/default/index.yaml [hunk 7]
```diff
@@ -1,7 +1,5 @@
 description: 'layout of ltplite for KPET'
-patterns:
-    -   pattern: .*
-        case_name: default/ltplite
+maintainer: maint1
+fetch_suffix: distribution/ltp/lite
 cases:
-    -   name: default/ltplite
-        tasks: suites/default/ltplite.xml
+    -   name: /kernel/distribution/ltp/lite
```
Comment (spbnick) [inline tests/assets/db/general/suites/default/index.yaml:6]: Please make this case name change a separate commit.

File: kpet/data.py [hunk 8]
```diff
@@ -213,6 +213,8 @@ def __init__(self, data):
                     match=Class(PositivePattern),
                     dont_match=Class(NegativePattern),
                     waived=Boolean(),
+                    role=String(),
+                    extra_params=String(),
```
Comment (spbnick) [inline kpet/data.py]: Even though `role` is Beaker-specific, we'll need it as is for now. However, we have a good opportunity to remove Beaker-dependence and obscurity with `extra_params`. Let's just call it `environment`, or `env_vars`, or something like that. It will be clear, to the point, and non-Beaker specific. Finally make it `Dict(String())` to represent variable names/values directly, instead of having to paste XML into YAML, where we don't have to. Make the default be an empty dictionary too to simplify templates.

**Review by spbnick — CHANGES_REQUESTED** [review]
Thank you, Jakub. This is going in the right direction, no question. However, I have a bunch of requests below, and inline.

Please keep pull request title and description up-to-date, here and on GitLab. It's difficult for me to recall which PR is which and what it does otherwise.

Please also keep logically-unrelated changes in separate commits (concerns kpet-db changes) and parts of the same logical change in a single commit. I.e. have "move waived param to yaml" and "move all params to yaml" in a single commit together. OTOH, have changing the tests-beaker URL, and case names in test assets separate from everything else and each other.

Also, please make sure every commit (not just every PR) works. In this case, the tests would stop working after the schema update, so the test change would need to be in the same commit with the schema change.

The above makes it easier for me to review/drop/merge/revert/think about specific commits.

File: kpet/data.py [hunk 9]
```diff
@@ -213,6 +213,8 @@ def __init__(self, data):
                     match=Class(PositivePattern),
                     dont_match=Class(NegativePattern),
                     waived=Boolean(),
+                    role=String(),
```
Comment (spbnick) [inline kpet/data.py:216]: Another thing: please assign the default `role` right here in the code, so you don't have to do it in the template and can just assume it always exists.

**Comment by spbnick** [comment]
And one more thing: please make sure the updated kpet works with kpet-db, both *with* and *without* your coming changes. That way we can update smoothly and safely.

File: kpet/data.py (hunk 2 above)
Comment (wizardofoz123) [reply kpet/data.py]: I keep thinking that every test should have a point of contact (some maintainer), never no-one. Wasn't this one of the requirements for onboarding?

Comment (spbnick) [reply kpet/data.py]: Weeell, to be honest, I'm not really sure about that. I think it's a good idea to make it required. However, we would need to add support for validating lists with a minimum number of elements (to be able to have a maintainer list and check that there's at least one). If you could do that, then I'm game :)

File: tests/assets/db/general/trees/rhel7.xml (hunk 4 above)
Comment (wizardofoz123) [reply tests/assets/db/general/trees/rhel7.xml]: I gather that you don't actually need any changes here. (And yes, the naming is funny, but that was purely by accident.)

Comment (spbnick) [reply tests/assets/db/general/trees/rhel7.xml]: Exactly, just sharing the amusing coincidence :smile:.

File: tests/assets/db/general/suites/default/index.yaml (hunk 7 above)
Comment (wizardofoz123) [reply tests/assets/db/general/suites/default/index.yaml:6]: The generated task name in XML doesn't actually change. The original has 
<task name="/kernel/distribution/ltp/lite" role="STANDALONE"> , so this change is here to make sure the output remains the same. Are you sure you want to split this?

Comment (spbnick) [reply tests/assets/db/general/suites/default/index.yaml:6]: Ah, if I got it right, and this is just bringing in the task's name, like all those changes in kpet-db, then sure, please keep it! Thanks :)

**Comment by wizardofoz123** [comment]
@cki-bot: test it.

**Comment by wizardofoz123** [comment]
The pipeline passed, there's only one test fail and that one is related to test itself, not this patch.
